### ISSUE TITLE
Add Device Protocol list in UI

### ISSUE THREAD

**Issue by gao270615179** [edgexfoundry/edgex-ui-go#373]
Add `Device Protocol` list in UI, because there will have two or more device protocols in some devices.

**Comment by badboy-huaqiao** [comment]
@gao270615179 Good Catch! 👍, that's also our plan not long ago, but due to some irresistible factors we would to do that next version.  

But if you have finished that section, we will be very glad to see that your code is committed here.

**Comment by badboy-huaqiao** [comment]
After the deeper research and discussion with core team, it doesn't looks like so necessary, what the basis of the conclusion is that which one should be used for device to create the connection at a time. for example, the modbus service supports both RTU and TCP, but only one of them will be used at a time. if there‘s dynamic switching needed, customize your own application.

**Comment by ajcasagrande** [comment]
This feature is useful for the `device-onvif-camera` service, as right now we utilize the ProtocolProperties as a way to store custom metadata about a device. The problem is that newly discovered devices get both the `Onvif` and `CustomMetadata` protocols added, but in the UI only the `CustomMetadata` shows up:

![Screenshot from 2022-07-05 11-22-00](https://user-images.githubusercontent.com/48735240/177391746-1ee62918-2e77-41cf-a2b5-312cc63fbf02.png)

![Screenshot from 2022-07-05 11-22-15](https://user-images.githubusercontent.com/48735240/177391761-a0f65b8e-ddda-46b6-b19f-2db60c0c35ea.png)

**Comment by badboy-huaqiao** [comment]
@ajcasagrande that would be great to support multiple device protocols, if we're planing to add new protocol template for `device-onvif-camera` service (see #531), this feature is required, assign it to me, I will update that.

**Comment by ajcasagrande** [comment]
@badboy-huaqiao This isn't required for #531 , though it would be nice. The device service will end up adding the missing `CustomMetadata` protocol I believe. This mostly effects the viewing and editing of existing devices.
